onnxruntime/core/framework/instant_op.cc

```diff
@@ -0,0 +1,273 @@
+    }
+  } else {
+    return Status(common::ONNXRUNTIME, common::FAIL, "Invalid attribute data type.");
+  }
+  *op_attr = reinterpret_cast<OrtOpAttr>(attr.release());
+  return Status::OK();
+}
+
+onnxruntime::Status CreateOperator(const OrtKernelInfo* info,
+                                   const char* op_name,
+                                   const char* domain,
+                                   int version,
+                                   const char** type_constraint_names,
+                                   const ONNXTensorElementDataType* type_constraint_values,
+                                   int type_constraint_count,
+                                   const OrtOpAttr* attr_values,
+                                   int attr_count,
+                                   OrtOp* op) {
+  *op = nullptr;
+  auto kernel_info = reinterpret_cast<const OpKernelInfo*>(info);
+  auto ep = reinterpret_cast<const IExecutionProvider*>(kernel_info->GetExecutionProvider());
+  auto kernel_registry = ep->GetKernelRegistry();
+  const KernelCreateInfo* kernel_create_info{};
+  std::unordered_map<std::string, MLDataType> type_constraint_map;
+  for (int i = 0; i < type_constraint_count; ++i) {
+    ONNX_NAMESPACE::TypeProto proto;
+    proto.mutable_tensor_type()->set_elem_type(type_constraint_values[i]);
+    type_constraint_map[type_constraint_names[i]] = DataTypeImpl::TypeFromProto(proto);
+  }
+  auto status = kernel_registry->TryFindKernel(op_name,
+                                               domain,
+                                               version,
+                                               type_constraint_map,
+                                               ep->Type(),
+                                               &kernel_create_info);
+  ORT_RETURN_IF_ERROR(status);
+  onnxruntime::Node node;
+  for (int i = 0; i < attr_count; ++i) {
+    auto attr_proto = reinterpret_cast<ONNX_NAMESPACE::AttributeProto*>(attr_values[i]);
+    node.AddAttributeProto(*attr_proto);
+  }
+  OpKernelInfo instant_kernel_info(node, KernelDef{}, *ep, {}, {}, {});
+  std::unique_ptr<onnxruntime::OpKernel> op_kernel;
+  FuncManager func_mgr;
+  status = kernel_create_info->kernel_create_func(func_mgr, instant_kernel_info, op_kernel);
+  ORT_RETURN_IF_ERROR(status);
+  *op = op_kernel.release();
+  return status;
+}
+
+onnxruntime::Status InvokeOperator(_In_ const OrtKernelContext* context,
+                                   _In_ const OrtOp ort_op,
+                                   _In_ const OrtValue* const* input_values,
+                                   _In_ int input_count,
+                                   _Inout_ OrtValue* const* output_values,
+                                   _In_ int output_count) {
+  auto ctx = reinterpret_cast<const OpKernelContext*>(context);
+  AllocatorPtr allocator{};
+  ORT_RETURN_IF_ERROR(ctx->GetTempSpaceAllocator(&allocator));
+  InstantKernelContext instant_ctx(input_values,
```

Comment:
In understand that we cannot use the custom op's `OpKernelInfo` to instantiate the wrapped ort kernel because the `node` changes (a comment explaining this in `CreateOperator()` will be nice),  but naively I would have thought we can use the `OrtKernelContext` instance (which is basically an `OpKernelInstance`) to pass in directly to the ort kernel. Not understanding the `InstantKernelContext` concept. Is there a design doc that I can use to understand this ?